packages/palette/src/elements/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,174 @@
+import React, {
+  Children,
+  createRef,
+  isValidElement,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react"
+import styled from "styled-components"
+import { useCursor } from "use-cursor"
+import { color, space } from "../../helpers"
+import { ChevronIcon } from "../../svgs"
+import { Box, BoxProps } from "../Box"
+import { Clickable } from "../Clickable"
+import { Skip } from "../Skip"
+import { VisuallyHidden } from "../VisuallyHidden"
+import { paginate } from "./utils"
+
+const Container = styled(Box)`
+  position: relative;
+  width: 100%;
+`
+
+const Viewport = styled(Box)`
+  width: 100%;
+  overflow: hidden;
+`
+
+const Rail = styled(Box)`
+  display: flex;
+  height: 100%;
+  transition: transform 500ms;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`
+
+const ARROW_WIDTH = space(4)
+
+const Arrow = styled(Clickable)`
+  position: absolute;
+  top: 0;
+  height: 100%;
+  width: ${ARROW_WIDTH}px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  user-select: none;
+  color: ${color("black30")};
+  transition: opacity 250ms, color 250ms;
+
+  > svg {
+    fill: currentColor;
+  }
+
+  &:hover,
+  &:focus {
+    outline: 0;
+    color: ${color("black100")};
+  }
+
+  &:disabled {
+    opacity: 0;
+    pointer-events: none;
+  }
+`
+
+const Next = styled(Arrow)`
+  right: 0;
+  transform: translateX(100%);
+`
+
+const Previous = styled(Arrow)`
+  left: 0;
+  transform: translateX(-100%);
+`
+
+const Cell = styled(Box)``
+
+export interface CarouselProps extends BoxProps {
+  children: JSX.Element | JSX.Element[]
+}
+
+/** Carousel */
+export const Carousel: React.FC<CarouselProps> = ({ children, ...rest }) => {
+  const viewportRef = useRef<HTMLDivElement | null>(null)
+  const startRef = useRef<HTMLButtonElement | null>(null)
+  const endRef = useRef<HTMLButtonElement | null>(null)
+
+  const cells = useMemo(
+    () =>
+      Children.toArray(children)
+        .filter(isValidElement)
+        .map(child => ({ child, ref: createRef<HTMLDivElement>() })),
+    []
+  )
```

Comment:
Wrap up children into a pageable array and create refs to each item which are threaded in when rendered. `useMemo` so that the refs are stable (`useRef` is frowned upon in `map` re: the rules of hooks).